File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    scoreDict = {}
    
    for i in range(len(name)):
        scoreDict[name[i]] = yearning[i]
        
    for p in photo:
        sum = 0
        for j in p:
            if j in scoreDict:
                sum += scoreDict[j]
        answer.append(sum)
        
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
import java.util.*;

class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        
        List<String> nameList = Arrays.asList(name);

        
        int index = 0;
        for (String[] p: photo) {
            for (int i=0; i<p.length; i++) {
                int pIndex = nameList.indexOf(p[i]);
                if (pIndex == -1) {
                    continue;
                }
                answer[index] += yearning[pIndex];
            }
            index++;
        }
        
        return answer;
    }
}